### PR TITLE
Allow multiple chests to be opened at once from items page

### PR DESCRIPTION
**Pull request by eonwarped**
Work in progress, while it works, it does not work ideally in that if you do not sign the transaction, and put in a different amount, the request will fail with 'previous order not confirmed'. Need to figure out a suitable way to handle this case, maybe from the API side.

### CODE REVIEW

File: src/components/RedeemCouponPopup.tsx [hunk 1]
```diff
@@ -1,38 +1,135 @@
...
+                value={amount}
+                onChange={e => setAmount(Number(e.target.value))}
+              />
+            </StyledInputContainer>
+            <Button Typography={Headline} width="100%" loading={submitted} onClick={onRedeem}>
+              {t('open')}
+            </Button>
+          </StyledRedeemContainer>
+        </StyledRedeemCouponPopup>
+      </Fullscreen>
```
Comment (pyc92) [inline src/components/RedeemCouponPopup.tsx:131]: Please extract this part of code into a component.

File: src/components/TreasuryChestPopup/index.tsx [hunk 2]
```diff
@@ -72,7 +72,9 @@ export default function TreasuryChestPopup() {
           </Button>
         </StyledContainer>
       </StyledFullscreen>
-      {redeem && treasuryChest && <RedeemCouponPopup coupon={treasuryChest} onClose={closeRedeemCouponPopup} />}
+      {redeem && treasuryChest && (
+        <RedeemCouponPopup coupon={treasuryChest} item={undefined} onClose={closeRedeemCouponPopup} />
```
Comment (pyc92) [inline src/components/TreasuryChestPopup/index.tsx:76]:
```suggestion
        <RedeemCouponPopup coupon={treasuryChest} onClose={closeRedeemCouponPopup} />
```

**Comment by 0xthk** [comment]
> Work in progress, while it works, it does not work ideally in that if you do not sign the transaction, and put in a different amount, the request will fail with 'previous order not confirmed'. Need to figure out a suitable way to handle this case, maybe from the API side.

This is by design. The user needs to use the signature first or he can re-roll again and again until he got the item he wants.

**Comment by eonwarped** [comment]
> > Work in progress, while it works, it does not work ideally in that if you do not sign the transaction, and put in a different amount, the request will fail with 'previous order not confirmed'. Need to figure out a suitable way to handle this case, maybe from the API side.
> 
> This is by design. The user needs to use the signature first or he can re-roll again and again until he got the item he wants.

Yeah, I figured it was part of the design. However,  it will make for a confusing UI experience if they do not remember the previous amount after not signing. Is there a way for the error to figure out what the previous tx signed was, or for the backend to remember the item order and give it in that order? Hmm the former is probably better.

File: src/components/RedeemCouponPopup.tsx (hunk 1 above)
Comment (eonwarped) [reply src/components/RedeemCouponPopup.tsx]: Done, check to see if this is this is what you have in mind

**Comment by 0xthk** [comment]
LGTM

**Pull request merged**